packages/medusa/src/api/routes/admin/customer-groups/list-customer-groups.ts

```diff
@@ -0,0 +1,87 @@
+import { IsNumber, IsOptional, IsString } from "class-validator"
+import { Type } from "class-transformer"
+import omit from "lodash/omit"
+
+import { validator } from "../../../../utils/validator"
+import { CustomerGroupService } from "../../../../services"
+import { CustomerGroup } from "../../../../models/customer-group"
+import { FindConfig } from "../../../../types/common"
+import { defaultAdminCustomerGroupsRelations } from "."
+
+/**
+ * @oas [get] /customer-groups
+ * operationId: "ListCustomerGroups"
+ * summary: "Retrieve a list of customer groups"
+ * description: "Retrieve a list of customer groups."
+ * x-authenticated: true
+ * parameters:
+ *   - (query) q {string} Query used for searching user group names.
+ *   - (query) offset {string} How many groups to skip in the result.
+ *   - (query) limit {string} Limit the number of groups returned.
+ * tags:
+ *   - CustomerGroup
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           properties:
+ *             customerGroup:
+ *               $ref: "#/components/schemas/customer_group"
+ */
+export default async (req, res) => {
+  const validated = await validator(AdminGetCustomerGroupsListParams, req.query)
+
+  const customerGroupService: CustomerGroupService = req.scope.resolve(
+    "customerGroupService"
+  )
+
+  let expandFields: string[] = []
+  if (validated.expand) {
+    expandFields = validated.expand.split(",")
+  }
+
+  const listConfig: FindConfig<CustomerGroup> = {
+    relations: expandFields.length
+      ? expandFields
+      : defaultAdminCustomerGroupsRelations,
+    skip: validated.offset,
+    take: validated.limit,
+    order: { created_at: "DESC" },
+  }
+
+  const filterableFields = omit(validated, ["limit", "offset", "expand"])
+
+  const [data, count] = await customerGroupService.listAndCount(
+    filterableFields,
+    listConfig
+  )
+
+  res.json({
+    count,
+    customerGroups: data,
+    offset: validated.offset,
+    limit: validated.limit,
+  })
+}
+
+export class AdminGetCustomerGroupsListParams {
+  @IsString()
+  @IsOptional()
+  q?: string
+
```

Comment:
We need to add a couple of things here:
```
id: string[]
name: string[]
created_at: DateComparisonOperator
updated_at: DateComparisonOperator
order: string
```

See [list-products.ts](https://github.com/medusajs/medusa/blob/c16df9383c08288f3643fde7aadad1becb2e4c9d/packages/medusa/src/api/routes/admin/products/list-products.ts) for reference.

All fields should ofc be optional